libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -571,32 +585,48 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 			else
 				args.emplace_back(convert(*arguments[i], *parameterTypes[i]).commaSeparatedList());
 
-		if (auto identifier = dynamic_cast<Identifier const*>(&_functionCall.expression()))
+		FunctionDefinition const* functionDef;
+		if (auto memberAccess = dynamic_cast<MemberAccess const*>(&_functionCall.expression()))
+		{
+			solUnimplementedAssert(!functionType->bound(), "Internal calls to bound library functions are not yet implemented");
+
+			functionDef = dynamic_cast<FunctionDefinition const*>(memberAccess->annotation().referencedDeclaration);
+			solAssert(functionDef, "Libraries cannot have state variables");
```

Comment:
I think the error message might be misleading - this does not have to be a library and the declaration can also be something else than a state variable.

Reply:
OK. I'll change it.

Reply:
Actually, I don't think it can be anything other than a call via a pointer or a direct function call. This code was inside the `switch` case for an internal call. I changed the message to `"Calls via pointers stored in state variables are not implemented"`. I hope it's a bit clearer.

Is there another Solidity construct that could fall into this switch case?

Also, now this case is possible because it could be a call via a function pointer stored in a state variable of a base contract so I changed it to `solUnimplementedAssert()`. I tried to enable it but it triggers an unimplemented exception somewhere in `YulUtilFunctions` so I guess it's really not implemented.

Reply:
I'm not sure what other cases could be possible, because of that I would just have left the message empty ;)

Reply:
The thing is, my code here is written with this assumption in mind. I want to make it easier for someone to call me out on it if I'm wrong. Or simply to see that my assumption was wrong and fix it.

Well, I'll try to turn this into a more specific assertion. This would make the message unnecessary.

Reply:
Please take a look now. It's now a more specific assertion without text so no risk of misleading anyone. If it's wrong, it will just fail.